bundles/org.eclipse.swt/Eclipse SWT PI/win32/library/os_custom.c

```diff
@@ -219,6 +227,10 @@ BOOL Validate_SetPreferredAppMode(const BYTE* functionPtr)
 		(functionPtr[0x00] == 0x8B) && (functionPtr[0x01] == 0x05) &&   // mov     eax,dword ptr [uxtheme!g_preferredAppMode]
 		(functionPtr[0x06] == 0x87) && (functionPtr[0x07] == 0x0D) &&   // xchg    ecx,dword ptr [uxtheme!g_preferredAppMode]
 		(functionPtr[0x0C] == 0xC3);                                    // ret
+#elif defined(_M_ARM64)
+	/* Not implemented yet */
+	functionPtr; /* to prevent: warning C4100: 'functionPtr': unreferenced formal parameter */
+	return FALSE;
```

Comment:
You should return TRUE here and SWT on Windows on ARM should require Windows 10 1903 or higher.

Or we can implement the disassembly hack that is done for x64 with ARM64 code.

If you need to understand why they did hacky detection, see
https://github.com/mintty/mintty/issues/983#issuecomment-611868330
and
https://stackoverflow.com/a/58547831/827532

Reply:
Yes, return TRUE solves the problem of scrollbars not in dark mode, on my WoA box! The scrollbars now show dark color correctly:
![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/49886739/ff8cf301-a64a-4396-9a45-35d3c02d532a)

I will fix the code later, thanks. Does that mean we don't need any disassembly hack for WoA then?

Reply:
If we state that SWT on Windows on ARM require Windows 1903+ then we can return TRUE there and not do the assembly hack.

Reply:
Also. I personally would've not done that assembly hack. If you look at one of the links above, this is way less hacky:
```
                auto ord135 = GetProcAddress(hUxtheme, MAKEINTRESOURCEA(135));
                if (g_buildNumber < 18334)
                    _AllowDarkModeForApp = reinterpret_cast<fnAllowDarkModeForApp>(ord135);
                else
                    _SetPreferredAppMode = reinterpret_cast<fnSetPreferredAppMode>(ord135);
```
In the Validate_AllowDarkModeForWindowWithTelemetryId() all they would have to do was to just query the build version of Windows to decide if TRUE or FALSE stating that uxtheme has the expected function on ordinal 135.

Reply:
I've updated the PR with new commit returning TRUE for Arm64, so no need for any further implementation. Windows on Arm64 is really only Windows 11 going forward, so there's no need for any checking for Windows 10 1903+ in Arm64 because Windows 10 is no longer supported in Arm64 by Microsoft.

Reply:
I think that what @SyntevoAlex is saying is that since these three functions are undocumented (they are not even exported by name, only by ordinal) they could be moved to another ordinal completely or even removed in the future builds of Windows 11 , thus it's unsafe to just return TRUE without checking first if the function is indeed the one expected at a given ordinal.

Reply:
Yes, the functions are still NOT documented, so they still need to be checked.

Reply:
@SyntevoAlex @hmartinez82 I don't understand any of the proposed changes to the `os_custom.c` file.

I'm thinking of removing this `os_custom.c` file from this PR, so that you guys can submit it in a separate new PR with your proposed changes for Arm64. And my PR will wait for your new PR to be submitted first.

Reply:
@chirontt I can do that, but I don't know how to compile this.  I have no Maven or Java experience :( 
That's why I copied the code here. I can try to explain to you were those numbers 0x18 and 0x1C are coming from:

The validation check for x64, was done by looking at the machine code of those three functions, `AllowDarkModeForWindow()`, `AllowDarkModeForWindowWithTelemetryId()`, `SetPreferredAppMode()`. 
This was done by someone one Visual Studio, in an x64 machine.
With the disassembly of the code, one can see which bytes are at each address after the start of the function (the start of the function is the value returned by GetProcAddress()

Here's `SetPreferredAppMode()` for instance on x64:
![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/1100440/375db533-2048-4b7b-903f-4d67fba18b59)

As you can see, you have to open the disassembly view to be able to Step In the function call (because we don't have the source code for uxtheme.dll), but it's doable. 
The numbers circled in blue in the picture above are being tested in this line for instance https://github.com/eclipse-platform/eclipse.platform.swt/pull/1045/files#diff-d6068e15e4497052c1129ba671192ebce3920a189d39a48d0cb66e151019f9b8R227 in `Validate_SetPreferredAppMode()`

Reply:
@SyntevoAlex @chirontt https://github.com/eclipse-platform/eclipse.platform.swt/pull/1048